lib/cloud_controller/opi/instances_client.rb

```diff
@@ -59,11 +58,52 @@ def desired_lrp_instance(process)
 
     private
 
-    def handle_error(response_body)
-      error = response_body['error']
-      return unless error
+    def process_has_no_desired_instances?(process)
+      process.stopped? || process.instances == 0
+    end
+
+    def confirm_not_running(process)
+      parsed_response = JSON.parse(client.get(instances_path(process)).body)
```

Comment:
What if the response is not JSON parseable? Like if Eirini is not reach able and you just get back like a `"no healthy upstream"` error from its Envoy sidecar?

Reply:
I'm wondering if we should have the `Retryable.retryable` block either in here to retry those cases or at a higher level up in the `lrp_instances` method and have both this and the `get_instances` method in its block.

Reply:
Oh lol ya, should prob actually handle errors here. The 1-2 levels above which are calling this are handling generic errors being raised, so they won't be lost entirely at least. Wrapping them to make them more clear would probably be good though.

If we put it in a retryable block though, the PR to limit retries which prompted this is kind of moot isn't it?

Reply:
The intent of the original PR was to not retry only on "not found"/404 style errors. It was because if your process was `STOPPED`/had zero instances this call would always retry five times and take five seconds to respond for every stopped app.

I think it was a mistake in the original PR that it didn't retry other classes of errors in the "no running instances" case.

Reply:
Ah ok, thanks for clarifying, that makes sense. Will adjust this accordingly then

Reply:
@tcdowney just pushed changes to match the original intent, seem fine?

Reply:
yes, I think this looks good 👍 

thanks! 🙂

Reply:
:tq: